groovy/gradle/src/org/netbeans/modules/gradle/GradleDataObject.java

```diff
@@ -174,6 +184,20 @@ static String getFileOrProjectName(FileObject primaryFile) {
         return ret;
     }
 
+    @ServiceProvider(service = MIMEResolver.class, position = -10)
+    public static class GradleKotlinMIMEResolver extends MIMEResolver {
```

Comment:
When I was working on the [performance team](http://wiki.netbeans.org/Performance) one of the goals we had was to replace all MIME resolvers with declarative ones. Introducing new programmatic ones goes against such effort.

Reply:
If this is still needed, it should be possible to tweak the default mime resolver to check the provided extensions and look at getNameExt() instead of just getExt() if the provided extension contains a dot.

Reply:
Could come up with a not perfect but good enough declarative mime type registration.